tests/selenium/pixel_tracking_test.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python
+# -*- coding: UTF-8 -*-
+
+import unittest
+import pbtest
+
+class pixelTrackingTesting(pbtest.PBSeleniumTest):
+	"""Tests for the cookie pixel tracking heuristic included in heuristicblocking.js
+		- loads gitcdn resource which places a tracking cookie on page then creates an img tag
+		- img tag makes a src request carrying a substring of that tracking cookie
+		- tracking domain is caught by pixel tracking heuristic, snitch map entry is updated
+	"""
+
+	def test_that_tracker_is_caught(self):
+		TESTING_URL = ("https://gitcdn.link/repo/ablanathtanalba/pixelTrackingTestingResource/master/rsrc.html")
```

Comment:
Might want to move this test to eff.org/files/ now as gitcdn.link is down again.

Reply:
The test succeeds even if we remove the contents of the pixel cooking sharing detection function from Privacy Badger prior to running the test. (The test doesn't fail when we need it to fail.) I think this is because the image resource being fetched comes from `github.com`, and it sets cookies. So our regular cookie tracking heuristic catches it.

Reply:
Hmm, I knew that github was already an entry in a pretrained snitch map, which is why I made the check in my test specifically for the inclusion of `gitcdn` in that particular entry, rather than just the existence of the entry itself... Good call on clearing out the pretrained data though, I'll add that in.

Reply:
Is this in reply to https://github.com/EFForg/privacybadger/pull/2425#discussion_r315423770?

Reply:
Related to both, yes

Reply:
I don't understand the connection to this thread.

Reply:
I misread you and thought that you were saying that `github.com` itself can't be checked for, as it already exists in the pretrained snitch_map, so I was clarifying that that's why I wrote it to explicitly check for the presence of `gitcdn` in the `github` entry. thanks for pointing that out!

Just waiting on resolution from Ops to give me write access, then I'll move my test file over to eff.org/files.

Reply:
Cool, gotcha. Plus, there is https://github.com/EFForg/privacybadger/pull/2425#discussion_r315421693, which is a larger issue. We need the test to fail when the functionality we mean to test is missing/broken, but the test succeeds because `github.com` sets cookies, and so we detect the `github.com` resource as a tracker but for the wrong reason.